### PR TITLE
fix build & github workflow

### PR DESCRIPTION
**Pull request by let-value**
## Description
Fixed prebuilds & added github workflow

## Motivation and Context
I was unable to install latest version on windows 
#292 #329 #325 #322  #321 

## How Has This Been Tested?
Package installed as a dependency from github package registry, and it worked

## Types of changes
* Fixed build.js
* Fixed install.js
* Accidentally formatted js files with prettier
* Created github workflow

### CODE REVIEW

**Comment by ash0x0** [comment]
This overlaps #312 for the github action. The build and install changes are however necessary. I suggest you remove the github action from this PR in favor of merging #312. If you see that there's an addition you can make to the action in #312 then do that as well please. Also, you removed package lock, make sure to commit the correct package lock for the package changes you made.

File: install.js
Comment (ash0x0) [inline install.js:31]: Ummmm... I don't think this is a good way to do it. Also, the project hasn't been maintained since 2012 and the github APIs change all the time so this doesn't really belong in a build script. Neither does the hardcoded one, but this solution doesn't make things better.

**Review by ash0x0 — CHANGES_REQUESTED** [review]
There are many style changes here and they're getting in the way of seeing the functional changes. If you'd like to make style changes (which I absolutely think you should) I think you should make that in a separate PR by adding lint + style packages and fix the entire repo. Right now to make things easier, I think you should remove all the style changes and keep only the functional changes.

File: install.js
Comment (let-value) [reply install.js:31]: This is the exact technique used when [uploading](https://github.com/prebuild/prebuild/blob/master/upload.js#L12) artifacts, I just reused it for downloading

Comment (ash0x0) [reply install.js]: In that case keep it as is.

File: build.js
Comment (let-value) [reply build.js]: Apologies for the day initially. Rolled back the style changes.

**Comment by ash0x0** [comment]
Do bring back package lock and check what you'd like to do about the action overlap with #312.

**Comment by let-value** [comment]
The lock file is in place, I didn't delete it.
My implementation solves a slightly different problem than #312. I not only make prebuilds but also make a scoped npm package for publishing to the GitHub repository. I just needed to build and use this project somewhere, and I did it in my fork. As you said, this project has not been supported for a long time and I did not hope for the merge of this request. Although I managed to contact the maintainers of this repository directly.

**Comment by ash0x0** [comment]
> The lock file is in place, I didn't delete it.
> My implementation solves a slightly different problem than #312. I not only make prebuilds but also make a scoped npm package for publishing to the GitHub repository. I just needed to build and use this project somewhere, and I did it in my fork. As you said, this project has not been supported for a long time and I did not hope for the merge of this request. Although I managed to contact the maintainers of this repository directly.

I actually meant (github-from-package)[https://github.com/substack/github-from-package] but I guess this one hasn't been maintained either.
I'm not sure who the current maintainer is but if you've managed to contact them please tell them to open an issue looking for maintainers and try to assign someone.

**Comment by WilixLead** [comment]
> > The lock file is in place, I didn't delete it.
> > My implementation solves a slightly different problem than #312. I not only make prebuilds but also make a scoped npm package for publishing to the GitHub repository. I just needed to build and use this project somewhere, and I did it in my fork. As you said, this project has not been supported for a long time and I did not hope for the merge of this request. Although I managed to contact the maintainers of this repository directly.
> 
> I actually meant [github-from-package](https://github.com/substack/github-from-package) but I guess this one hasn't been maintained either.
> I'm not sure who the current maintainer is but if you've managed to contact them please tell them to open an issue looking for maintainers and try to assign someone.

At this moment I'm maintainer (but still high loaded in primary job) this is PR ready to merge?

**Comment by ash0x0** [comment]
> At this moment I'm maintainer (but still high loaded in primary job) this is PR ready to merge? 
If you're busy I think it'd be a good idea to add a couple of people with write access so we can have more active maintenance on the project. If you're up to that I think you should open an issue to check who's interested.

This PR shouldn't be merged until the action is modified. I think you should merge #312 and then come back to this after @let-value modified it to be merged.
@let-value I think you should remove the action and merge the rest given that the action was actually meant for fork publishing.

File: install.js
Comment (ash0x0) [inline install.js:33]:
```suggestion
  const url = pkg.homepage;
```
This won't work with your base but once rebased will work.

File: package.json
Comment (ash0x0) [inline package.json:38]: No need for this now with the suggestion to `install.js`

**Review by ash0x0 — CHANGES_REQUESTED** [review]

**Comment by ash0x0** [comment]
Do please change the name of ci workflow as well so it won't cause conflicts when merging this with #312 as both are good and I will be cherry-picking after merge.

**Pull request merged**
